code/go/0chain.net/smartcontract/storagesc/allocation.go

```diff
@@ -1080,14 +1080,14 @@ func (sc *StorageSmartContract) updateAllocationRequestInternal(
 
 	// an allocation can't be shorter than configured in SC
 	// (prevent allocation shortening for entire period)
-	if request.Expiration < 0 &&
+	if newExpiration < 0 ||
 		newExpiration-t.CreationDate < toSeconds(conf.MinAllocDuration) {
 
 		return "", common.NewError("allocation_updating_failed",
 			"allocation duration becomes too short")
 	}
 
-	if request.Size < 0 && alloc.Size+request.Size < conf.MinAllocSize {
+	if alloc.Size+request.Size < conf.MinAllocSize || request.Size+alloc.Size < alloc.UsedSize {
```

Comment:
The `alloc.Size+request.Size` could be assigned to a variable and to make it cleaner.